### ISSUE TITLE
[Feature Request]: Steal focus only when window is active

### ISSUE THREAD

**Issue by acerbastimur** [electron/electron#33450]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Hi all,

I am developing a clipboard app and I need a feature like described below:

- User has a focus and cursor blinking in input on (for ex.) the Whatsapp app
- When the user runs the clipboard app, it should be stealing the focus without losing the condition on the Whatsapp app
- When I quit the clipboard app, the focus proceeds to blink in the Whatsapp app

I am not sure if there is a covering solution to steal the focus for the case electron app is opened, and let the focus proceed where it left off.

### Proposed Solution

There is a method to steal focus programmatically:

`      app.focus({ steal: true });
`

This feature might be added to this part. (Like instead of having a boolean, we might have an enum option)

### Alternatives Considered

I tried to change this parts of my code but I got nothing :(

  let level = "normal";
  // Mac OS requires a different level for our drag/drop and overlay
  // functionality to work as expected.
  if (process.platform === "darwin") {
    level = "floating";
  }
  mainWindow.setAlwaysOnTop(true, level);
  mainWindow.webContents.openDevTools({ mode: 'undocked' });

**Comment by BB-fat** [comment]
Maybe you should try this:
```
  mainWindow.setAlwaysOnTop(true, 'floating');
  mainWindow.setVisibleOnAllWorkspaces(true, { visibleOnFullScreen: true });
  if (isMac()) {
    mainWindow.setWindowButtonVisibility(false);
  }
```

**Comment by abpbackup** [comment]
@acerbastimur you are right. `app.focus({ steal: true })` was working fine to do so in 22.2.1. It seems that got broken on 23.x.x and fixed again on 24.0.0.beta-x
